### PR TITLE
BUG: Fix scalar range step in qMRMLModelDisplayNodeWidget

### PR DESCRIPTION
**Pull request by agirault**
Should have been using log10 and not log
Use significantDecimals and closestPowerOfTen from ctk

### CODE REVIEW

**Comment by agirault** [comment]
Failing check from CircleCI:
CMake Error at /usr/src/Slicer-build/EMSegment/Testing/CMakeLists.txt:45 (include):
  include could not find load file:

    MIDAS

```
CMake Error at /usr/src/Slicer-build/EMSegment/Testing/CMakeLists.txt:644 (midas_add_test):
  Unknown CMake command "midas_add_test".
```

**Comment by agirault** [comment]
CircleCI check failure seems unrelated:
```
-- Configuring remote module: EMSegment
-- Configuring SEM CLI module: EMSegmentCommandLine
-- Configuring SEM CLI module: IslandRemoval
-- Configuring SEM CLI module: EMSegmentTransformToNewFormat
-- Looking for decorator header qSlicerEMSegmentModuleWidgetsPythonQtDecorators.h
-- Looking for decorator header qSlicerEMSegmentModuleWidgetsPythonQtDecorators.h - Not found
-- Configuring Loadable module: EMSegment [qSlicerEMSegmentModuleExport.h]
-- Configuring Loadable module: EMSegmentQuick [qSlicerEMSegmentQuickModuleExport.h]
CMake Error at /usr/src/Slicer-build/EMSegment/Testing/CMakeLists.txt:45 (include):
  include could not find load file:

    MIDAS


CMake Error at /usr/src/Slicer-build/EMSegment/Testing/CMakeLists.txt:644 (midas_add_test):
  Unknown CMake command "midas_add_test".


-- Configuring incomplete, errors occurred!
```

**Comment by jcfr** [comment]
> CircleCI check failure seems unrelated:

You can ignore this. We are in the process of fixing the CI.

File: Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx [hunk 1]
```diff
@@ -633,16 +633,23 @@ void qMRMLModelDisplayNodeWidget::updateWidgetFromMRML()
...
   int decimals = 0;
-  if (displayRange[0]<displayRange[1])
+  if (displayRange[0] < displayRange[1])
     {
+    // Begin with a precision of 1% of the range
     precision = displayRange[1]/100.0 - displayRange[0]/100.0;
+    // Extend min/max by 20% to give some room
     newMin = (floor(displayRange[0]/precision) - 20 ) * precision;
     newMax = (ceil(displayRange[1]/precision) + 20 ) * precision;
+    // Use closest power of ten value as a step value
```
Comment (jcfr) [inline Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: Consider using `ctk::closestPowerOfTen`
See https://github.com/commontk/CTK/blob/master/Libs/Core/ctkUtils.h#L114

Comment (jcfr) [reply Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: Adn for the `decimals`, would it work to use `ctk::significantDecimals` ?

Comment (agirault) [reply Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: Nice, that behavior is unwanted though:
https://github.com/commontk/CTK/blob/master/Libs/Core/ctkUtils.h#L109

Comment (agirault) [reply Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: done

Comment (agirault) [reply Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: let me investigate that more, could make a good use of `significantDecimals`

Comment (agirault) [reply Modules/Loadable/Models/Widgets/qMRMLModelDisplayNodeWidget.cxx]: I think I have a satisfying solution now

**Comment by agirault** [comment]
revision [r27101](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=27101)

**Comment by jcfr** [comment]
Thanks @agirault  :+1:

**Pull request closed without merging**